diff -up kbd-2.4.0/src/setfont.c.orig kbd-2.4.0/src/setfont.c
--- kbd-2.4.0/src/setfont.c.orig	2020-06-02 11:30:41.000000000 +0200
+++ kbd-2.4.0/src/setfont.c	2021-05-17 12:06:44.360616147 +0200
@@ -189,8 +189,8 @@ int main(int argc, char *argv[])
 	if (restore)
 		kfont_restore_font(kfont, fd);
 
-	if (ifilct)
-		kfont_load_fonts(kfont, fd, ifiles, ifilct, iunit, hwunit, no_m, no_u);
+	if (ifilct && (ret = kfont_load_fonts(kfont, fd, ifiles, ifilct, iunit, hwunit, no_m, no_u)) < 0)
+		return -ret;
 
 	if (ufil && (ret = kfont_load_unicodemap(kfont, fd, ufil)) < 0)
 		return -ret;
